aliases.csv

```diff
@@ -117,6 +117,11 @@ fr_fo,fr@formal
 fr_form,fr@formal
 fr_fr_x_formal,fr@formal
 fr_fr_x_informal,fr@informal
+#,Punjabi Shahmukhi (Perso-Arabic script) codes in use
+lah,pa_PK
```

Comment:
Are you sure these are same? See https://github.com/WeblateOrg/language-data/blob/main/languages.csv#L330

Reply:
Yes. Lahnda is not a real language; the name comes from the linguist George Grierson from British colonial times who thought Punjabi was two languages. (Parsi is another example of a non-existing language having a code - Parsis are an ethnicity in Gujarat who speak Gujarati.) There are a series of Punjabi dialects known as Lehndi, not Lahnda. Lahnda means "west," but it has to inflect for the grammatical gender, and languages are feminine. So we can say Lahnda Punjab and Charda Punjab to describe West Punjab and East Punjab, but we have to say Lehndi and Chardi Punjabi to describe the dialects spoken in these regions. Hence why you will never actually hear anyone say Lahnda to describe a language, as the name comes from a misunderstanding of the grammar.

Some examples of texts by Punjabi linguists which discuss Lehndi, as a dialect group, not a separate language:
* https://clrjournal.com/ojs/index.php/clrjournal/article/view/10
* https://bharatavani.in/bharatavani/punjabi/book?post_category=book&id=%E0%A8%B2%E0%A8%B9%E0%A8%BF%E0%A9%B0%E0%A8%A6%E0%A9%80%20%E0%A8%AA%E0%A9%B0%E0%A8%9C%E0%A8%BE%E0%A8%AC%E0%A9%80%20%E0%A8%A6%E0%A8%BE%20%E0%A8%B6%E0%A8%AC%E0%A8%A6%20%E0%A8%95%E0%A9%8B%E0%A8%B6%20(%E0%A8%9A%E0%A9%81%E0%A8%97%E0%A8%B5%E0%A9%87%E0%A8%82%20%E0%A8%B6%E0%A8%AC%E0%A8%A6)%20%7C%20Lehndi%20Punjabi%20Da%20Shabad%20Kosh%20(Chugven%20Shabad)

A common feature of texts which use the term Lahnda is they only cite English-language sources.

I can't imagine an actual use for this code to mean a particular dialect, or anything other than just a synonym for Punjabi Pakistan, because dialect is to fluid for this kind of distinction to make sense for encoding a locale. My family is from Western Punjab/Lahnda Punjab but we use Doabi and Majhi which are considered Eastern/Chardi or Central/Kendri dialects because much of Pakistani Punjab was actually populated by people from Jalandhar and thereabouts in the 19th century...

I can provide other info if necessary; I'm not sure if that explains things clearly.